data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -141,26 +141,31 @@ public void initialize() {
     public Optional<SourcePartition<T>> getNextPartition(final Function<Map<String, Object>, List<PartitionIdentifier>> partitionCreationSupplier) {
         validateIsInitialized();
 
-        if (partitionManager.getActivePartition().isPresent()) {
-            return partitionManager.getActivePartition();
-        }
+        Optional<SourcePartitionStoreItem> ownedPartitions = Optional.empty();
+        try {
+          if (lock.tryLock()) {
```

Comment:
This lock is still blocking the `tryAcquireAvailablePartition` call unnecessarily. Please change to align with my comment below.